speaker/layer2_controller.go

```diff
@@ -135,8 +139,14 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 }
 
 func (c *layer2Controller) SetBalancer(l log.Logger, name string, lbIPs []net.IP, pool *config.Pool) error {
+	ifs := c.announcer.GetInterfaces()
 	for _, lbIP := range lbIPs {
 		ipAdv := ipAdvertisementFor(lbIP, c.myNode, pool.L2Advertisements)
+		if !ipAdv.MatchInterfaces(ifs...) {
+			level.Warn(l).Log("op", "SetBalancer", "protocol", "layer2", "service", name, "IPAdvertisement", ipAdv,
+				"localIfs", ifs, "msg", "the specified interfaces used to announce LB IP don't exist")
+			return ErrSpecifiedInterfacesNotExist
```

Comment:
if we return here, we won't announce the other ips on the list.
What I would do is inject the client to layer2Controller and throw the event from here, without returning

Reply:
Because of the init order of speaker.Main, we can't inject the client to Layer2Controller. So I inject the client by func SetBalancer's param.